### PR TITLE
Only restrict to image files when onlyImages is active

### PR DESCRIPTION
**Pull request by arknu**
Fixes U4-8034  (http://issues.umbraco.org/issue/U4-8034)

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/views/common/overlays/mediaPicker/mediapicker.controller.js [hunk 1]
```diff
@@ -16,7 +16,12 @@ angular.module("umbraco")
             $scope.startNodeId = dialogOptions.startNodeId ? dialogOptions.startNodeId : -1;
             $scope.cropSize = dialogOptions.cropSize;
             $scope.lastOpenedNode = $cookieStore.get("umbLastOpenedMediaNodeId");
-            $scope.acceptedFileTypes = mediaHelper.formatFileTypes(Umbraco.Sys.ServerVariables.umbracoSettings.imageFileTypes);
+            if($scope.onlyImages){
+                $scope.acceptedFileTypes = mediaHelper.formatFileTypes(Umbraco.Sys.ServerVariables.umbracoSettings.imageFileTypes);
+            }
+            else {
+                $scope.acceptedFileTypes = "*.*";
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/views/common/overlays/mediaPicker/mediapicker.controller.js]: Should be: 

`mediaHelper.formatFileTypes(Umbraco.Sys.ServerVariables.umbracoSettings.disallowedUploadFiles).replace(/\./g, "!.");`

This is just like other uploaders, allow all files except for the ones in the `disallowedUploadFiles` list. 
I know this is handled server side as well, but might as well do it correctly here too.

Comment (arknu) [inline src/Umbraco.Web.UI.Client/src/views/common/overlays/mediaPicker/mediapicker.controller.js]: I only discovered the fact that you can do a negative accepted list yesterday, quite useful actually. Was going to update my PR this evening, but you beat me to it ☺ Or should I still push an update this evening?

Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/views/common/overlays/mediaPicker/mediapicker.controller.js]: Yup, would be great if you have a moment please!

Comment (arknu) [inline src/Umbraco.Web.UI.Client/src/views/common/overlays/mediaPicker/mediapicker.controller.js]: Will do it this evening then ☺ (in about 3-4 hours)

**Comment by arknu** [comment]
@nul800sebastiaan I have updated the PR with your requested changes. I also rebased it to the latest dev-v7 and tested in conjunction with the "Only images" option, now that it works. Everything works as expected, i.e. when "Only images" is checked, you can only upload image files, otherwise you can upload file (as long it is not a disallowed type).

**Comment by nul800sebastiaan** [comment]
Awesome, thanks very much!

**Pull request merged**